packages/rum-core/src/performance-monitoring/transaction-service.js

```diff
@@ -280,13 +286,23 @@ class TransactionService {
            * and once performance observer is disconnected
            */
           if (tr.captureTimings) {
-            if (metrics.tbt.duration > 0) {
-              tr.spans.push(createTotalBlockingTimeSpan(metrics.tbt))
+            const { cls, fid, tbt } = metrics
+            if (tbt.duration > 0) {
+              tr.spans.push(createTotalBlockingTimeSpan(tbt))
+            }
+
+            tr.experience = {}
+            if (isPerfTypeSupported(LONG_TASK)) {
+              tr.experience.tbt = tbt.duration
+            }
+
+            if (isPerfTypeSupported(LAYOUT_SHIFT)) {
+              tr.experience.cls = cls
```

Comment:
can you declare these metrics in the metrics object prior to setting it? Two things
1. makes it easier to track
2. object is in same shape, performance optimisation in engines.

Reply:
Let's add fid to the metrics